streams/shared/src/main/scala/zio/stream/StreamEffect.scala

```diff
@@ -322,19 +331,22 @@ private[stream] object StreamEffect extends Serializable {
       }
     }
 
-  final def fromInputStream(is: InputStream): StreamEffect[Any, Nothing, Byte] =
-    StreamEffect[Any, Nothing, Byte] {
+  final def fromInputStream(
+    is: InputStream,
+    chunkSize: Int = ZStreamChunk.DefaultChunkSize
+  ): StreamEffect[Any, IOException, Chunk[Byte]] =
+    StreamEffect[Any, IOException, Chunk[Byte]] {
       Managed.effectTotal {
-        def pull(): Byte = {
-          val a = is.read()
-          if (a < 0) {
-            end
-          } else {
-            a.toByte
-          }
+        val buf = Array.ofDim[Byte](chunkSize)
```

Comment:
The `buf` has to be re-created every time `pull` runs, or we'll be mutating chunks that have already been emitted.

Reply:
Updated this